cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java

```diff
@@ -454,4 +458,37 @@ protected static <R> R connectionResult(DSLContext dslContext, ConnectionCallabl
         }
     }
 
+    protected String parseLocFromTimeSeriesId(String tsId) {
+        String[] parts = tsId.split("\\.");
+        return parts[0];
+    }
+
+    @NotNull
+    protected VersionType getVersionType(String names, String office, boolean dateProvided) {
+        VersionType dateVersionType;
+
+        if (!dateProvided) {
+            boolean isVersioned = isVersioned(names, office);
+
+            if (isVersioned) {
+                dateVersionType = VersionType.MAX_AGGREGATE;
+            } else {
+                dateVersionType = VersionType.UNVERSIONED;
+            }
+
+        } else {
+            dateVersionType = VersionType.SINGLE_VERSION;
+        }
+
+        return dateVersionType;
+    }
+
+    private boolean isVersioned(String names, String office) {
+        return connectionResult(dsl, connection -> {
```

Comment:
open an issue about caching this in memory.

Reply:
also this should be in timeseries dao not this low level

Reply:
moved into TimeSeriesDaoImpl

Reply:
https://github.com/USACE/cwms-data-api/issues/607